### PR TITLE
Add mvc for booking

### PR DESCRIPTION
**Pull request by ritamma**

### CODE REVIEW

File: app/controllers/bookings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  def show; end
+
+  def new
+    @booking = current_user.bookings.new
+  end
+
+  def create
+    @booking = current_user.bookings.new(booking_params)
+    @booking.end_at -= 1
```
Comment (fooooxmr) [inline app/controllers/bookings_controller.rb]: Move to before_validation callback.

File: app/controllers/bookings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if @booking.save
+      total_duration_set
+      total_price_set
```
Comment (fooooxmr) [inline app/controllers/bookings_controller.rb]: Move to callbacks.

File: app/models/booking.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  def valid_start_date
+    return unless start_at < DateTime.now
+
+    errors.add(:start_at, "You can't choose date before today")
+  end
+
+  def booking_time
+    total_d = ((end_at - start_at) / 60).ceil
+    if total_d < 60 && service.name != 'Message chair'
```
Comment (fooooxmr) [inline app/models/booking.rb]: Move logic to service.

File: app/models/booking.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      errors.add(:end_at, 'The minimum booking time is 1 hour.')
+    elsif total_d < 15 && service.name == 'Message chair'
+      errors.add(:end_at, 'The minimum booking time is 15 minutes.')
+    end
+  end
+
+  def check_the_possibility_to_book
+    if service_id == 1 && Booking.with_service(1).reservations_overlap(end_at, start_at).count >= 5 ||
```
Comment (fooooxmr) [inline app/models/booking.rb]: Do not use IDs inside validations.

Comment (fooooxmr) [reply app/models/booking.rb]: Add count of possible bookings to service.

File: app/models/service.rb [hunk 5]
```diff
@@ -1,8 +1,9 @@
 class Service < ApplicationRecord
   belongs_to :company
+  has_many :bookings, dependent: :destroy
```
Comment (fooooxmr) [inline app/models/service.rb]: Do not destroy bookings.
Archive service, not destroy.

File: app/models/user.rb [hunk 6]
```diff
@@ -4,6 +4,7 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable,
          :omniauthable, omniauth_providers: [:google_oauth2]
+  has_many :bookings, dependent: :destroy
```
Comment (fooooxmr) [inline app/models/user.rb]: Archive users not destroy.

File: db/migrate/20230419090150_create_bookings.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class CreateBookings < ActiveRecord::Migration[7.0]
+  def change
+    create_table :bookings do |t|
+      t.datetime :start_at
+      t.datetime :end_at
+      t.integer :total_price
```
Comment (fooooxmr) [inline db/migrate/20230419090150_create_bookings.rb]: make total_price float

File: db/seeds.rb [hunk 8]
```diff
@@ -1,7 +1,56 @@
...
+company = Company.create!(
+  id: 1,
+  name: 'Legenda',
+  address: 'Repina-38',
+  phone: '+375291234567',
+  description: 'Billiard club in Grodno'
+)
+
+Service.create!([{
+                   id: 1,
```
Comment (fooooxmr) [inline db/seeds.rb]: remove ID from seeds.

**Review by fooooxmr — CHANGES_REQUESTED** [review]

File: app/controllers/bookings_controller.rb (hunk 1 above)
Comment (ritamma) [reply app/controllers/bookings_controller.rb]: Done

File: app/models/booking.rb (hunk 3 above)
Comment (ritamma) [reply app/models/booking.rb]: Done

File: app/models/booking.rb (hunk 4 above)
Comment (ritamma) [reply app/models/booking.rb]: Done

File: app/models/user.rb (hunk 6 above)
Comment (ritamma) [reply app/models/user.rb]: Done

File: db/seeds.rb (hunk 8 above)
Comment (ritamma) [reply db/seeds.rb]: Done

**Review by fooooxmr — APPROVED** [review]

**Pull request merged**